File: backend/aitrace-java-backend/src/main/java/com/supertrace/aitrace/domain/Project.java
package com.supertrace.aitrace.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "project")
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_uuid", nullable = false)
    private UUID userId;

    @NotBlank
    private String name;

    private String description;

    private String strategy;

    @Column(name = "avg_duration", nullable = false)
    private Integer averageDuration;

    @NotNull
    private BigDecimal cost;

    @Column(name = "created_timestamp", nullable = false)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdTimestamp;

    @Column(name = "last_update_timestamp", nullable = false)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime lastUpdateTimestamp;

    @PrePersist
    protected void onCreate() {
        this.createdTimestamp = LocalDateTime.now();
    }

    public Project refreshLastUpdateTimestamp() {
        this.lastUpdateTimestamp = LocalDateTime.now();
        return this;
    }

}

File: kubent/src/repository/models/project.py
from uuid import UUID
from datetime import datetime
from sqlalchemy import String, DateTime, Integer, Numeric, Text, text
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.dialects.postgresql import UUID as PG_UUID
from .base import Base

class Project(Base):
    __tablename__ = "project"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    user_uuid: Mapped[UUID] = mapped_column(PG_UUID(as_uuid=True), nullable=False)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    created_timestamp: Mapped[datetime] = mapped_column(DateTime(timezone=False), nullable=False, server_default=text("CURRENT_TIMESTAMP"))
    avg_duration: Mapped[int] = mapped_column(Integer, default=0, server_default=text('0'), nullable=False)
    cost: Mapped[float] = mapped_column(Numeric, default=(0.00), server_default=text('0'), nullable=False)
    last_update_timestamp: Mapped[datetime] = mapped_column(DateTime(timezone=False), nullable=False, server_default=text("CURRENT_TIMESTAMP"))
    description: Mapped[str | None] = mapped_column(String(255), nullable=True)
    strategy: Mapped[str | None] = mapped_column(Text, nullable=True)